pkg/autodiscovery/autoconfig.go

```diff
@@ -203,6 +203,12 @@ func (ac *AutoConfig) LoadAndRun() {
 	log.Debug("LoadAndRun done.")
 }
 
+// ForceRanOnceFlag sets the ranOnce flag.  This is used for testing other
+// components that depend on this value.
+func (ac *AutoConfig) ForceRanOnceFlag() {
+	atomic.StoreUint32(&ac.ranOnce, 1)
+}
```

Comment:
This is kind of yucky, but also hopefully temporary -- this RanOnce support is here only for the logs agent, and only to solve a race condition around container_collect_all.  Once that support is moved into AD itself, that race should be resolved and we can remove this.